File: frontend/src/types/course.ts
export type CourseType = 'ZORUNLU' | 'SECMELI';
export type Semester = 'GUZ' | 'BAHAR' | 'YAZ_OKULU';

export type CourseResponse = {
  id: string;
  code: string;
  name: string;
  facultyId: string;
  facultyName: string;
  departmentId: string;
  departmentName: string;
  academicianId: string;
  academicianName: string;
  theoreticalHours: number;
  practicalHours: number;
  ects: number;
  credits: number;
  courseType: CourseType;
  semester: Semester;
  grade: number;
  active: boolean;
};

export type CourseListResponse = {
  courses: CourseResponse[];
};

export type CreateCourseRequest = {
  code: string;
  name: string;
  facultyId: string;
  departmentId: string;
  academicianId: string;
  theoreticalHours: number;
  practicalHours: number;
  ects: number;
  credits: number;
  courseType: CourseType;
  semester: Semester;
  grade: number;
  active: boolean;
};

export type UpdateCourseRequest = CreateCourseRequest;

export type AcademicianResponse = {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  title: string;
  facultyId: string;
  facultyName: string;
  departmentId: string;
  departmentName: string;
};

File: backend/src/main/java/com/dts/dersliktakip/entity/Course.java
package com.dts.dersliktakip.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "courses")
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true, length = 20)
    private String code;

    @Column(nullable = false, length = 255)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "faculty_id", nullable = false)
    private Faculty faculty;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "academician_id", nullable = false)
    private Academician academician;

    @Column(name = "theoretical_hours", nullable = false)
    private int theoreticalHours = 2;

    @Column(name = "practical_hours", nullable = false)
    private int practicalHours = 0;

    @Column(nullable = false)
    private int ects = 3;

    @Column(nullable = false)
    private int credits = 2;

    @Column(name = "student_count", nullable = false)
    private int studentCount = 0;

    @Enumerated(EnumType.STRING)
    @Column(name = "course_type", nullable = false, length = 20)
    private CourseType courseType = CourseType.ZORUNLU;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Semester semester = Semester.GUZ;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "academic_period_id", nullable = false)
    private AcademicPeriod academicPeriod;

    @Column(nullable = false)
    private int grade = 1;

    @Column(nullable = false)
    private boolean active = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now();
    }
}
